include/zephyr/pm/state.h

```diff
@@ -154,6 +154,14 @@ struct pm_state_info {
 	 * @note 0 means that this property is not available for this state.
 	 */
 	uint32_t exit_latency_us;
+
+#ifdef CONFIG_PM_STATE_CUSTOM_DATA
+	/**
+	 * Target specific custom data. Support for runtime customization of
+	 * a particular idle state.
+	 */
+	const void *custom_data;
```

Comment:
Not a fan of the "stuff a void* into a struct" paradigm; untyped APIs are dangerous.   In situations where we have a need for platform-specific additional data in core kernel sturcts, zephyr typically adds a substruct, e.g. something like:

```
IF_ENABLED(CONFIG_SOC_HAS_PM_STATE, struct soc_pm_state soc);
```